tiledb/sm/filesystem/posix.cc

```diff
@@ -347,6 +347,26 @@ Status Posix::copy_file(
   return Status::Ok();
 }
 
+Status Posix::copy_dir(
+    const std::string& old_path, const std::string& new_path) {
+  create_dir(new_path);
+  std::vector<std::string> paths;
+  ls(old_path, &paths);
+  while (!paths.empty()) {
+    std::string file_name_abs = paths.front();
+    std::string file_name = file_name_abs.substr(old_path.length() + 1);
+    paths.erase(paths.begin());
+    if (is_dir(file_name_abs)) {
+      create_dir(new_path + "/" + file_name);
+      ls(file_name_abs, &paths);
```

Comment:
This is interesting. It definitely works because `ls()` appends to `paths`, but I would actually expect `ls()` to set rather than append.

I would suggest doing this instead, for safety:
```
std::vector<std::string> child_paths;
ls(file_name_abs, &child_paths);
paths.insert(paths.end(), child_paths.begin(), child_paths.end());
```